checker/bin/wpi.sh

```diff
@@ -0,0 +1,174 @@
+    exit 4
+fi
+
+function configure_and_exec_dljc {
+
+  if [ -f build.gradle ]; then
+      if [ -f gradlew ]; then
+    chmod +x gradlew
+    GRADLE_EXEC="./gradlew"
+      else
+    GRADLE_EXEC="gradle"
+      fi
+      CLEAN_CMD="${GRADLE_EXEC} clean -g .gradle -Dorg.gradle.java.home=${JAVA_HOME}"
+      BUILD_CMD="${GRADLE_EXEC} clean compileJava -g .gradle -Dorg.gradle.java.home=${JAVA_HOME}"
+  elif [ -f pom.xml ]; then
+      if [ -f mvnw ]; then
+    chmod +x mvnw
+    MVN_EXEC="./mvnw"
+      else
+    MVN_EXEC="mvn"
+      fi
+      # if running on java 8, need /jre at the end of this Maven command
+      if [ "${JAVA_HOME}" = "${JAVA8_HOME}" ]; then
+          CLEAN_CMD="${MVN_EXEC} clean -Djava.home=${JAVA_HOME}/jre"
+          BUILD_CMD="${MVN_EXEC} clean compile -Djava.home=${JAVA_HOME}/jre"
+      else
+          CLEAN_CMD="${MVN_EXEC} clean -Djava.home=${JAVA_HOME}"
+          BUILD_CMD="${MVN_EXEC} clean compile -Djava.home=${JAVA_HOME}"
+      fi
+  elif [ -f build.xml ]; then
+    # TODO: test these more thoroughly
+    CLEAN_CMD="ant clean"
+    BUILD_CMD="ant clean compile"
+  else
+      echo "no build file found for ${REPO_NAME}; not calling DLJC"
+      WPI_RESULTS_AVAILABLE="no"
+      return
+  fi
+
+  DLJC_CMD="${DLJC} -t wpi $* -- ${BUILD_CMD}"
+
+  if [ ! "x${TIMEOUT}" = "x" ]; then
+      TMP="${DLJC_CMD}"
+      DLJC_CMD="timeout ${TIMEOUT} ${TMP}"
+  fi
+
+  # Remove old DLJC output.
+  rm -rf dljc-out
+
+  # ensure the project is clean before invoking DLJC
+  eval "${CLEAN_CMD}" < /dev/null
+
+  echo "${DLJC_CMD}"
+
+  # This command also includes "clean"; I'm not sure why it is necessary.
+  eval "${DLJC_CMD}" < /dev/null
+
+  if [[ $? -eq 124 ]]; then
+      echo "dljc timed out for ${DIR}"
+      WPI_RESULTS_AVAILABLE="no"
+      return
+  fi
+
+  if [ -f dljc-out/wpi.log ]; then
+      WPI_RESULTS_AVAILABLE="yes"
+  else
+      WPI_RESULTS_AVAILABLE="no"
+  fi
+}
+
+#### Check and setup dependencies
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+# clone or update DLJC
+if [ -d "${SCRIPTDIR}/.do-like-javac" ]; then
+    git -C "${SCRIPTDIR}/.do-like-javac" pull
+else
+    git -C "${SCRIPTDIR}" clone https://github.com/kelloggm/do-like-javac --depth 1 .do-like-javac || (echo "Cannot clone do-like-javac" && exit 1)
+fi
+
+DLJC="${SCRIPTDIR}/.do-like-javac/dljc"
+
+#### Main script
+
+pushd "${DIR}" || exit 1
+
+configure_and_exec_dljc "$@"
+
+if [ "${WPI_RESULTS_AVAILABLE}" = "no" ]; then
+      # if running under Java 11 fails, try to run
+      # under Java 8 instead
+    export JAVA_HOME="${JAVA8_HOME}"
+    echo "couldn't build using Java 11; trying Java 8"
+    configure_and_exec_dljc "$@"
+    export JAVA_HOME="${JAVA11_HOME}"
+fi
+
+# support wpi-many.sh's ability to delete projects without usable results
+# automatically
+if [ "${WPI_RESULTS_AVAILABLE}" = "no" ]; then
+    echo "dljc could not run the build successfully"
```

Comment:
Please add output telling the user where to look for diagnostic output.
I think the answer is "in the `dljc-out` directory".